ios-coding-guidelines.md

```diff
@@ -46,4 +46,14 @@ These guidelines are meant to be used without Interface Builder and teach you ho
 
 ## Xcode
 
-1. **No White-Spaces In Group/Folder Names** Scripts do not like whitespaces. Since we have a lot of automation by scripts, please use `_` as replacement for them.
\ No newline at end of file
+1. **No White-Spaces In Group/Folder Names** Scripts do not like whitespaces. Since we have a lot of automation by scripts, please use `_` as replacement for them.
+
+## Accessibility
+
+1. * Use the accessibility id of a view to make it identifiable for unit tests or voice over announcements. The syntax for those identifiers should be `aid.viewController.view` or if further scoping is needed `aid.viewController.parentView.view`.
+2. * Set the appropriate accessibility traits if not set automatically. While buttons usually have it inherently set, labels functioning as headers need need their trait set to `.header`. The same applies for search fields. In cases were another view serves as a button, it's trait should be set to `.button`.
+3. * Use post notifications to inform about screen changes, for example ride cancellation, disappearing or appearing popovers or warning labels.
+4. * Images should either be disabled or have a description meaningful enough to be worth a blind person's time.
+5. * Focussable elements should be kept to the necessary minimum. For example, labels and controls that belong together should be treated as one element. The element should have its label's text for its a11y label and the control as its action. (Compare the title + switch cells in the iOS Settings App.)
+5. * Any colors should have a sufficient contrast (minimum 3:1).
```

Comment:
Should be min. 4.5:1 for fonts <18pt and min. 3:1 for fonts >=18pt

Since our main fonts body3 and body4 are 17pt and 15pt this is an important info. You can link the [official docs](https://www.w3.org/TR/WCAG20/#visual-audio-contrast-contrast)